src/Engine/GameAction.java

```diff
@@ -99,7 +99,6 @@ public GameEventQueue getEvents(MapMaster gameMap)
         isValid &= (gameMap.getLocation(attackLocation).getResident() == defender);
         isValid &= (null != defender) && attacker.canAttack(defender.model, attackRange, moved);
         isValid &= (null != defender) && attacker.CO.isEnemy(defender.CO);
-        isValid &= (null == moveLocation.getResident()) || (attacker == moveLocation.getResident());
```

Comment:
Infinite recursion?

Reply:
Yeah. I hit a case multiple times with Muriel vs Spender in FoW where each one would be trying to perform an attack through FoW, but the move tiles were both hidden. This check failed, so the action was cancelled. If just one AI hits the case it's OK(ish); it just ends the turn (though it should ambush the unit instead). If both hit the case, they ping-pong back and forth without generating any actual events to execute. Execution goes from `MapController.animationEnded` to `SpriteMapView.animate` and back again repeatedly, with occasional trips to `MapController.startNextTurn` and back to `animate` until it runs out of stack.

Reply:
I'm OK just removing this check, because this is exactly what `Utils.enqueueMoveEvent` handles.

Reply:
Reading this again, yes, the check is actually wrong and makes the program not work correctly.
I believe it would also prevent a human player from attempting to attack from a square that an enemy is on.
Might be worth throwing checks for this in the fog test/making sure this isn't checked in other events..